### ISSUE TITLE
Unable to replicate GDAL output in Python

### ISSUE THREAD

**Issue by ivanhigueram** [OSGeo/gdal#2801]
# Issue/Bug

I have been using `gdal` to transform a set of `GRIB` files to `GeoTIFF` and warp them to change the longitude from (0, 360)  to (-180, 180). In a nutshell, I use both `gdal_translate` and `gdal_warp` to accomplish this:  

```bash 
gdal_translate -of GTiff -b 1 test_grib.grib2 test_grib.tif
gdalwarp -t_srs WGS84 test_grib.tif test_grib_warp.tif \
  -wo SOURCE_EXTRA=1000 \
  --config CENTER_LONG 0 -overwrite
```

Once the process has finish I get a file with the following *global* stats: 

```bash
gdalinfo test_grib_warp.tif--stats 
[A lot of data that I'm ignoring here]
    STATISTICS_MAXIMUM=46.135308837891
    STATISTICS_MEAN=8.1518501020584
    STATISTICS_MINIMUM=-57.964697265625
    STATISTICS_STDDEV=20.985407239038
```

Now, I wanted to replicate this same behavior in Python using the `osgeo` library. I skipped the translation part since I realize `osgeo.gdal` can warp the GRIB file directly instead of having to convert/translate to a GeoTIFF format. For that I used the following Python code: 

```python
import os
from osgeo import gdal

filename = " test_grib.grib2"
file_temp_path = "test_grib_warp_python.tif"

OPTS = gdal.WarpOptions(dstSRS='WGS84',
                        warpOptions=['SOURCE_EXTRA=1000'],
                        options=['CENTER_LONG 0'])

try:
   ds = gdal.Open(filename)
except RuntimeError:
   ds = gdal.Open(str(filename))

if os.path.getsize(filename):
   ds_transform = gdal.Warp(file_temp_path,
                             ds,
                             options=OPTS)
   # is this a hack?
   ds_transform = None
else:
   print(f'{filename} is an empty file. No GDAL transform')
```

If I get the *global* statistics for this file I get completely different data: 

```bash
gdalinfo test_grib_warp_python.tiff--stats 
[A lot of data that I'm ignoring here]
    STATISTICS_MAXIMUM=46.135308837891
    STATISTICS_MEAN=11.468900010467
    STATISTICS_MINIMUM=-57.964697265625
    STATISTICS_STDDEV=17.677681479367
```

I have compared agains other data sources and the `gdal` result from the command-line is the one closest to reality, while the Python one is a little bit biased. I have tried different options from Python and the results seem to be quite the same. Is this an implementation problem on my end, or there are differences between `osgeo.gdal` and the `gdal` from the command-line. 

## System version

The gdal from the command line is: `GDAL 2.1.2, released 2016/10/24`, and my `osgeo` version is: ` '2.4.4'`. I'm using a Miniconda environment with Python 3.8. This is also posted [StackOverflow][1]

### Data
You can use the [following file][2] to replicate the results above

  [1]: https://stackoverflow.com/questions/62780414/unable-to-replicate-gdal-output
  [2]: https://www.dropbox.com/s/sdpashlt7fhv23p/gdas.t18z.pgrb2.0p25.f003.grib2?dl=1

**Comment by rouault** [comment]
> options=['CENTER_LONG 0']

This is not the right way of specifying a configuration option. Remove that options keyword and use
``gdal.SetConfigOption('CENTER_LONG', '0')``

**Comment by ivanhigueram** [comment]
Thanks for your reply @rouault . 

I tried the following: 

```python
import os
from osgeo import gdal

filename = " test_grib.grib2"
file_temp_path = "test_grib_warp_python.tif"

gdal.SetConfigOption('CENTER_LONG', '0')

OPTS = gdal.WarpOptions(dstSRS='WGS84',
                        warpOptions=['SOURCE_EXTRA=1000'])

try:
   ds = gdal.Open(filename)
except RuntimeError:
   ds = gdal.Open(str(filename))

if os.path.getsize(filename):
   ds_transform = gdal.Warp(file_temp_path,
                             ds,
                             options=OPTS)
   # is this a hack?
   ds_transform = None
else:
   print(f'{filename} is an empty file. No GDAL transform')
```

But the results are the same, the local statistics are off by several orders of magnitude.

**Comment by rouault** [comment]
Please provide a link to a file that can be used to replicate the issue

**Comment by rouault** [comment]
Ignore previous message

**Comment by ivanhigueram** [comment]
Dear Even,

A file is attached to the issue at the end of the body text, it's a Dropbox
link.

Thanks for your help

Iván

**Comment by rouault** [comment]
I've just tried with latest GDAL master and I get the same result through gdalwarp and your above Python snippet. I can't think of any particular reason for which this wouldn't work with 2.4.4 too, but maybe something has been fixed or there was a bug in a version after 2.1.2 that was fixed later. Anyway closing as works for me
